lib/wanda/messaging.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Wanda.Messaging do
+  @moduledoc """
+  Publishes messages to the message bus
+  """
+
+  @spec publish(String.t(), any()) :: :ok | {:error, any()}
+  def publish(topic, message), do: adapter().publish(topic, message)
```

Comment:
Could we maybe use the `routing_key` wording here?
Rather than using `routing_key` and `topic`

Reply:
Routing key is an internal of rabbitmq, it doesn't exist in other contexts/message brokers. The common term to define this is "topic", so from the abstract messaging implementation perspective, which doesn't know about the underlying adapter, topic sounds more fitting to me.

Reply:
Ok. I got confused with the fact that the "rabbitmq` adapter is named "amqp", and the terminologies are a bit mixed (routing_key is used in some places and topic in other, what rabbitmq is not referred anywhere.
Maybe the adapter should be named `rabbitmq`?
Anyway, I see that you already had a lot of thinking on this, so I don't want to extend it more.
It is fine as it is.